test/test_mod.c

```diff
@@ -225,5 +225,5 @@ int main(int argc, char **argv) {
     bam_destroy1(b);
     sam_hdr_destroy(h);
     hts_base_mod_state_free(m);
-    return 1;
+    return sam_close(in) != 0 ? 1 : 2;
```

Comment:
You need to deal with the case where the `sam_close()` on line 216 fails.

Reply:
Fixed to avoid attempting to close again if it failed the first time.